### DISCUSSION TITLE
Templated webids

### DISCUSSION THREAD

**Discussion by damooo** [sebadob/rauthy#257] [Ideas]
Hello @sebadob 

Currently webids are coupled with user-ids. Thus they are not configurable by service providers on how they look.

Is it possible to take a template string on config like `https://example.org/<$user_name>#me`, and take optional user_name input from user at the signup, and make resultant webid also one of independent primary keys in table?

Thank you.

**Comment by sebadob** [comment]
I don't really get what you mean with that.  
You want to get rid of the user id and replace it with a user-chosen nickname?

**Reply by damooo** [reply]
Webids are to be relatable (printable on tea shirts/cards for example.) Hence, we can provide them with this option. You can see, for eg [use.id](https://get.use.id/people) service. webid will be like `https://use.id/<$user>`.

**Reply by sebadob** [reply]
This would not really work.  
Rauthy's implementation of webid's should provide the basics, but it is not and never will be any kind of social media platform or provider. This is simply out of Rauthy's scope.
If you want a feature-packed webid hosting, it would be really easy to write a simple application which does this in less than a day.

What you could also do is to proxy requests. `webid` lookups should follow redirects. This means you could set up any DNS / URL you like and just return an HTTP 301 with the location pointing at the webid from Rauthy, if you want to use it. This way you could get rid of the API prefix as well, if you want to.

**Reply by damooo** [reply]
Yes, but the webid in token response cannot be modified?

**Reply by sebadob** [reply]
> Yes, but the webid in token response cannot be modified?

That's correct.  
But if you want to print your webid on your shirt, you could print the pretty one and everything would work with redirects.

**Reply by damooo** [reply]
The webid is "uri string". The string literal is significant. It is what get matched by apps.

**Reply by damooo** [reply]
Is there any plan to consider this @sebadob ? Thank you.

**Reply by sebadob** [reply]
I need to think about it a bit, if its possible to get a nice solution without "destroying" anything else.  
But it has a low priority. I am working on dynamic client registration currently, then I have a few small other features on the TODO and want to release v0.21.  
I can think about a possible solution for the version afterwards.
